mono/metadata/icall.c

```diff
@@ -1017,33 +1017,24 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_SufficientExecutionStac
 #elif defined(TARGET_ANDROID)
 	// No need for now
 #else
-	guint8 *stack_addr;
-	guint8 *current;
-	size_t stack_size;
-	int min_size;
-	MonoInternalThread *thread;
-
-	mono_thread_info_get_stack_bounds (&stack_addr, &stack_size);
-	/* if we have no info we are optimistic and assume there is enough room */
-	if (!stack_addr)
-		return TRUE;
+	guint8 *stack_start;
+	guint8 *stack_current;
+	gsize stack_size;
+	MonoThreadInfo *info;
+
+	info = mono_thread_info_current ();
+	stack_start = info->stack_start_limit;
+	stack_size = info->stack_end - info->stack_start_limit;
 
-	thread = mono_thread_internal_current ();
-	// .net seems to check that at least 50% of stack is available
-	min_size = thread->stack_size / 2;
+	stack_current = (guint8 *)&stack_start;
 
-	// TODO: It's not always set
-	if (!min_size)
+	/* Some platforms support dynamic stack resizing */
+	if (stack_current < stack_start)
```

Comment:
This is not correct, even on platforms that support growable stacks (windows) they have a bounded max size and we compute that in mono_thread_info_get_stack_bounds

Reply:
The linux implementation is also slow for the main thread at least, i.e.:
https://bugzilla.xamarin.com/show_bug.cgi?id=58911